### DISCUSSION TITLE
Support for n number of function arguments

### DISCUSSION THREAD

**Discussion by matthew-dean** [grain-lang/grain#1790] [Ideas]
I'm not sure if this is included yet, but I didn't see it in function documentation.

It seems like it would be essential to be able to define functions that can have any number of arguments, like:

```
let fn = (...args) => {
  // do things with each of the args (in a tuple?)
}
```

**Comment by spotandjake** [comment]
It's not a language feature yet it's been discussed before. It is not supported by a lot of languages though just as a note most languages that do not have it you just pass a list through to the function instead. 



There are some complications with Variadic functions in Wasm as well codegen wise which could be worked around but it isn't a feature that will probably be landing anytime soon. As far as I know there has been no outright no to adding them eventually though. I could be wrong on that though.
